tests/src/JIT/Generics/Constraints/Nullable.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Reflection;
+
+public class Test
+{
+    class G<T, U> where T : U
+    {
+    }
+
+    static object s_x;
+
+    static int Main()
+    {
+        try
+        {
+            typeof(G<,>).MakeGenericType(typeof(int), typeof(int?));
+
+            Console.WriteLine("Test Failed1");
+            return 1;
+        }
+        catch (ArgumentException)
+        {
+        }
+
+        ///////////////////////////////////
+        // Test object cast
+
+        s_x = (object)1234;
+
+        if (!(s_x is int?) || (int?)s_x != 1234)
+        {
+            Console.WriteLine("Test Failed2");
+            return 1;
+        }
+
+        ///////////////////////////////////////
+        ///  Test reflection
+
+        Type nubInt = typeof(Nullable<int>);
+        Type intType = typeof(int);
+        Type objType = typeof(object);
+        Type valTypeType = typeof(ValueType);
+
+        // sanity checks
+        // Nullable<T>  is assignable from  int
+        if (!nubInt.IsAssignableFrom(intType))
```

Comment:
It would be better to add these tests to CoreFX. CoreFX has better infrastructure for API testing like this.